synchronizer/synchronizer.go

```diff
@@ -315,6 +333,31 @@ func (s *Synchronizer) AdvanceView(syncInfo hotstuff.SyncInfo) {
 		}
 	}
 
+	b, ok := syncInfo.Block()
+	if !ok {
+		b = s.leafBlock
+	}
+
+	highQCBlock, ok := s.blockChain.LocalGet(s.highQC.BlockHash())
+	if !ok {
+		s.logger.Error("AdvanceView: Could not find block referenced by new QC!")
```

Comment:
Is it safe to continue after this? Needless to say, if `highQCBlock` is nil, `Extends()` below will give a nil pointer panic.

Reply:
Fixed in 600a0f7.